extension/packages/nextjs/app/eip-712/page.tsx

```diff
@@ -0,0 +1,154 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import Link from "next/link";
+import type { NextPage } from "next";
+import useSWRMutation from "swr/mutation";
+import { SignTypedDataReturnType } from "viem/accounts";
+import { useAccount, useSignTypedData, useVerifyTypedData } from "wagmi";
+import {
+  EIP_712_DOMAIN,
+  EIP_712_TYPE,
+  VerifyRequestBody,
+  generateMessageToBob,
+} from "~~/utils/eip-712";
+import { notification } from "~~/utils/scaffold-eth";
+import { postMutationFetcher } from "~~/utils/swr";
+
+const Home: NextPage = () => {
+  const { address: connectedAddress } = useAccount();
+  const [signature, setSignature] = useState<SignTypedDataReturnType>();
+  const { signTypedDataAsync } = useSignTypedData();
+
+  const [name, setName] = useState("");
+  const [message, setMessage] = useState("");
+
+  const { trigger: verify } = useSWRMutation(
+    "/api/verify",
+    postMutationFetcher<VerifyRequestBody>
+  );
```

Comment:
Umm I think we should probably remove `swr` and use native `fetch` directly

`swr` seems add a bit of complexity atleast for beginners and also bit overpowered for this example. 

We actually can use tanstack query (already installed in SE-2) to get pending / loading state. But we are not using those states here any which ways and also makes sense not to use them to keep things simple and minimal

So I would vote for just using simple fetch POST request

Reply:
+1

Reply:
I agree, as I remember I added it for consistency with some other example. But I can't find it :D.

Removed [981741a](https://github.com/scaffold-eth/create-eth-extensions/pull/3/commits/981741a88e885d1ec1be0d103f325a5bbbbe1211)